### PR TITLE
STA process documentation

### PR DESCRIPTION
**Pull request by rabbitstack**
This PR explains which metrics and metadata is collected for monitored processes. It also touches on how STA masks sensitive data in process's command line args.

### CODE REVIEW

File: docs/agents/sematext-agent/processes/metrics.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+title: Process Monitoring
+description: Process Monitoring memory CPU usage
+
+Sematext Agent gathers various metrics and metadata about running processes and brings them to [Sematext Cloud](https://sematext.com/cloud/) so you can painlessly explore the system resources allocated by processes whether they're executing on bare metal machines or scheduled to run on multi-cluster Kubernetes workloads.
+
+Process collector is enabled by default. You can disable the collection of the process's metrics/metadata by setting the following properties:
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: You can disable the collection of the process's metrics/metadata by => You can disable process metrics/metadata collection by

File: docs/agents/sematext-agent/processes/metrics.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   </pre>
+ </div>
+ <div class="mdl-tabs__panel" id="env-enabled">
+   <pre>
+   PROCESS_ENABLED=false
+   </pre>
+ </div>
+</div>
+
+Similarly, to tweak the frequency of the process's data series delivery to Sematext Cloud adjust the following options:
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: process's => process'

File: docs/agents/sematext-agent/processes/metrics.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ <div class="mdl-tabs__tab-bar">
+     <a href="#file-interval" class="mdl-tabs__tab is-active">Config file</a>
+     <a href="#env-interval" class="mdl-tabs__tab">Env variable</a>
+ </div>
+
+ <div class="mdl-tabs__panel is-active" id="file-interval">
+   <pre>
+   # st-agent.yml
+   process:
+     interval: 10s
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: Too bad we didn't pick a better, more descriptive name for this.  A person not reading the documentation carefully could easily think this is what controls how often process information is _collected_, when in fact this is controlling how often the collected data is _shipped_.

File: docs/agents/sematext-agent/processes/metrics.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+ <div class="mdl-tabs__panel" id="env-interval">
+   <pre>
+   PROCESS_INTERVAL=10s
+   </pre>
+ </div>
+</div>
+
+### Metrics
+
+The agent collects top _N_ processes by CPU and memory utilization. Once process's is elected as a top resource consumer, it stays in the top N list for the time slice that has been allocated to it. It is relinquished from the top N list whether there is another process with higher resource usage that displaces the current process or it leaves the top N list voluntarily (e.g. no longer detected as a relevant resource consumer).
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: Once process's is => Once process is
whether there is => when there is

File: docs/agents/sematext-agent/processes/metrics.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+The table below includes all available metrics.
+
+| Metric name       | Description |
+| ------------------|-------------|
+| resident memory   | Represents the resident memory allocated by the process. It comprises the memory allocated for heap and stack areas |
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: Full stop missing
comprises => includes (simpler word for some)

File: docs/agents/sematext-agent/processes/metrics.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+| CPU usage         | Designates the process's CPU utilization in percentage       |
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: Designates the process's CPU utilization in percentage => Designates the process' percentage of CPU utilization

File: docs/agents/sematext-agent/processes/metrics.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+The agent collects top _N_ processes by CPU and memory utilization. Once process's is elected as a top resource consumer, it stays in the top N list for the time slice that has been allocated to it. It is relinquished from the top N list whether there is another process with higher resource usage that displaces the current process or it leaves the top N list voluntarily (e.g. no longer detected as a relevant resource consumer).
+
+The table below includes all available metrics.
+
+| Metric name       | Description |
+| ------------------|-------------|
+| resident memory   | Represents the resident memory allocated by the process. It comprises the memory allocated for heap and stack areas |
+| CPU usage         | Designates the process's CPU utilization in percentage       |   
+| fd count          | Is the number of file descriptors allocated by the process     |
+| thread count      | Represents the number of threads running in the process's address space  |
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: process's => process'

File: docs/agents/sematext-agent/processes/metrics.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+| disk read bytes   | Is the amount of bytes that the process fetched from the storage layer     |
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: amount of bytes => number of bytes

File: docs/agents/sematext-agent/processes/metrics.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+| disk write bytes  | Is the amount of bytes which the process sent to the storage layer     |
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/metrics.md]: amount of bytes => number of bytes

File: docs/agents/sematext-agent/processes/sensitive-data.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+The agent ensures any sensitive data such as credentials or tokens are scrubbed before shipping the data to Sematext Cloud.
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/sensitive-data.md]: the data => data

File: docs/agents/sematext-agent/processes/sensitive-data.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+
+For example, launching a Python process with `--password` and `MYSQL_PASSWORD` arguments passed to it, would result in the command line metadata attribute being masked as following:
```
Comment (otisg) [inline docs/agents/sematext-agent/processes/sensitive-data.md]: Do/should we list all patterns we look for?  If we don't cover something that one uses, can a person add his/her pattern somewhere?

Comment (rabbitstack) [reply docs/agents/sematext-agent/processes/sensitive-data.md]: We can list the patterns, but currently we don't support custom patterns :(

Comment (otisg) [reply docs/agents/sematext-agent/processes/sensitive-data.md]: +1 for listing patterns and customer patterns is something we can consider adding IF we see demand for this.

File: docs/agents/sematext-agent/processes/metrics.md (hunk 7 above)
Comment (rabbitstack) [reply docs/agents/sematext-agent/processes/metrics.md]: Just curious. Is this considered a typo or just a preferred way to form possessives? I've read people using "process`s" frequently.

Comment (otisg) [reply docs/agents/sematext-agent/processes/metrics.md]: When a word ends with "s" then you just append the apostrophe.  e.g. "This is Otis' comment".

Comment (rabbitstack) [reply docs/agents/sematext-agent/processes/metrics.md]: Thanks for explaining.

File: docs/agents/sematext-agent/processes/metrics.md (hunk 3 above)
Comment (rabbitstack) [reply docs/agents/sematext-agent/processes/metrics.md]: This actually controls how often process data is collected, but as soon as collection stage completes, data is shipped as well. So, I would look at this like a single unit of work where data shipping happens right after collection is done.

**Comment by otisg** [comment]
Should this be merged to master so I can deploy these docs?

**Comment by rabbitstack** [comment]
I think we can do that. Want me to push the merge to master?

**Pull request merged**